nexus/src/external_api/views.rs

```diff
@@ -75,14 +75,50 @@ impl From<model::Project> for Project {
 
 // IMAGES
 
-/// Client view of Images
+/// Client view of global Images
+#[derive(ObjectIdentity, Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct GlobalImage {
+    #[serde(flatten)]
+    pub identity: IdentityMetadata,
+
+    /// URL source of this image, if any
+    pub url: Option<String>,
+
+    /// Version of this, if any
+    pub version: Option<String>,
+
+    /// Hash of the image contents, if applicable
+    pub digest: Option<Digest>,
+
+    /// size of blocks in bytes
+    pub block_size: ByteCount,
+
+    /// total size in bytes
+    pub size: ByteCount,
+}
+
+/// Client view of project Images
 #[derive(ObjectIdentity, Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct Image {
     #[serde(flatten)]
     pub identity: IdentityMetadata,
 
-    pub project_id: Option<Uuid>,
+    /// The project the disk belongs to, if not a global image.
+    pub project_id: Uuid,
```

Comment:
The "if" seems a little odd here - these must always belong to projects, no?

Reply:
you're right, removed in e4007d47